runtime/src/tests.rs

```diff
@@ -723,6 +817,70 @@ mod pallets_interact {
             });
         }
     }
+
+    mod sudo {
+        use super::*;
+
+        #[test]
+        fn test_set_sudo_to_multisig() {
+            new_test_ext().execute_with(|| {
+                // Extract the account IDs from SAMPLE_USERS
+                let account_ids: Vec<AccountId32> = testsfixtures::SAMPLE_USERS
+                    .iter()
+                    .map(|user| AccountId32::from(user.raw_account))
+                    .collect();
+                let multi = Multisig::multi_account_id(&account_ids[..3], 2);
+
+                assert_ok!(Balances::transfer_allow_death(
+                    RuntimeOrigin::signed(account_ids[0].clone()),
+                    MultiAddress::Id(multi.clone()),
+                    1000
+                ));
+
+                // Check existing sudo key
+                let existing_sudo_key = Sudo::key();
+                assert_eq!(existing_sudo_key, Some(account_ids[0].clone()));
+
+                // Setting the multisig account as the new sudo account
+                assert_ok!(Sudo::set_key(
+                    RuntimeOrigin::signed(account_ids[0].clone()),
+                    MultiAddress::Id(multi.clone())
+                ));
```

Comment:
Minuscule thing here. The following is more robust wrt the specific runtime configuration?
```suggestion
                // Setting the multisig account as the new sudo account
                assert_ok!(Sudo::set_key(
                    RuntimeOrigin::signed(existing_sudo_key.expect("No sudo key")),
                    MultiAddress::Id(multi.clone())
                ));
```